cpp/CMakeLists.txt

```diff
@@ -106,39 +111,23 @@ include(cmake/thirdparty/get_rmm.cmake)
 include(cmake/thirdparty/get_libcudacxx.cmake)
 include(cmake/thirdparty/get_cuco.cmake)
 
-if(BUILD_TESTS)
+if(BUILD_TESTS OR RAFT_ENABLE_NN_COMPONENT)
```

Comment:
WIth this PR, I notice FAISS is always building from source now, rather than linking to an existing shared lib if it exists.